### PR TITLE
feat: Gcox/update splitter

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
## Description

Update to use NPM package of react-splitter.  The latest version improves performance, relative split layout correctness, and customization of the splitter.  Events are also supported.

## Task Item

#minor

## Screenshots

No visible UI changes

### CODE REVIEW

File: Composer/packages/client/src/components/Split/ThinSplitter.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+const HitArea = styled.div(({ horizontal, hoverColor }: { horizontal: boolean; hoverColor: string }) => ({
+  boxSizing: 'border-box',
+  outline: 'none',
+  overflow: 'hidden',
+  height: '100%',
+  width: '100%',
+  cursor: horizontal ? 'row-resize' : 'col-resize',
+  background: 'transparent',
+  '%:hover .thin-split-visual': {
```
Comment (hatpick) [inline Composer/packages/client/src/components/Split/ThinSplitter.tsx:21]: is this valid syntax? % ? shouldn't it be &?

File: Composer/packages/client/src/components/Split/ThinSplitter.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    background: `${hoverColor}`,
```
Comment (hatpick) [inline Composer/packages/client/src/components/Split/ThinSplitter.tsx:22]: hoverColor is string itself.

File: Composer/packages/client/src/components/Split/ThinSplitter.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+/**
+ * The default splitter which provides a thin line within a possibly larger mouse hit area.
+ * @param props
+ */
+export const ThinSplitter = (props: Props) => {
+  const { horizontal, pixelSize, color, hoverColor } = props;
+
+  return (
+    <HitArea horizontal={horizontal} hoverColor={hoverColor}>
+      <Splitter className="thin-split-visual" color={color} horizontal={horizontal} splitterSize={pixelSize} />
```
Comment (hatpick) [inline Composer/packages/client/src/components/Split/ThinSplitter.tsx:51]: I suggest using a variable for _thin-split-visual_.

File: Composer/packages/client/src/components/Split/ThinSplitter.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    </HitArea>
+  );
+};
+
+export const renderThinSplitter = (props: RenderSplitterProps) => {
+  return (
+    <ThinSplitter
+      {...props}
+      color={props.dragging ? 'black' : 'silver'}
```
Comment (hatpick) [inline Composer/packages/client/src/components/Split/ThinSplitter.tsx:60]: why not pass dragging to the styled component and decide the colors within?

**Review by hatpick — APPROVED** [review]

File: Composer/packages/client/src/components/Split/ThinSplitter.tsx (hunk 1 above)
Comment (GeoffCoxMSFT) [reply Composer/packages/client/src/components/Split/ThinSplitter.tsx:21]: Yeah - that is weird.  Will fix.

File: Composer/packages/client/src/components/Split/ThinSplitter.tsx (hunk 3 above)
Comment (GeoffCoxMSFT) [reply Composer/packages/client/src/components/Split/ThinSplitter.tsx:51]: Emotion won't let me use a format string in the &hover style as the property name.

**Review by hatpick — DISMISSED** [review]

**Review by hatpick — APPROVED** [review]

**Pull request merged**